### PR TITLE
Adding documentation for vscode-dotnet-runtime extension

### PR DESCRIPTION
**Pull request by sfoslund**
## Summary

Adding docs for our new VSCode extension: https://github.com/dotnet/vscode-dotnet-runtime

Note: this link (https://github.com/dotnet/vscode-dotnet-runtime/blob/master/Documentation/commands.md) doesn't work yet but I have a PR out that will add this page: https://github.com/dotnet/vscode-dotnet-runtime/pull/148

### CODE REVIEW

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (wli3) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:3]: ?

**Comment by wli3** [comment]
Other than the TODO looks good to me. Although I think doc team have the final call

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (sfoslund) [reply docs/core/additional-tools/vscode-dotnet-runtime.md]: Leftover, thanks for catching that :)

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (NTaylorMullen) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:2]:
```suggestion
title: .NET Install Tool for VSCode Extension Authors
```

Not sure how these docs will be surfaced but sprinkling "VSCode" in front of extension authors felt more applicable since there's lots of types of extensions for various platforms.

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (NTaylorMullen) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:26]: oooo I didn't know they adopted this already too 🥳

**Review by NTaylorMullen — APPROVED** [review]

**Review by tdykstra — COMMENTED** [review]
Would this be a better fit in the [Visual Studio Code extensions documentation](https://code.visualstudio.com/api)?

**Comment by sfoslund** [comment]
> Would this be a better fit in the Visual Studio Code extensions documentation?

I'm not sure, I'm not able to find any specific extensions documented there, only the API itself.

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (Youssef1313) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:24]:
```suggestion
- [Azure Resource Manager (ARM) Tools for Visual Studio Code](https://github.com/microsoft/vscode-azurearmtools)
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

**Comment by tdykstra** [comment]
@gregvanl Is there an appropriate place in the VS Code docs for this article?

**Comment by gregvanl** [comment]
@tdykstra This extension looks to be still under development and given the caveat in the README about reaching out via the GitHub repo, I don't think we'd want to add this topic to the VS Code documentation yet. I also think it might be better to keep the documentation with the extension source as this point so it is easier for the engineers to update and rely on the extension README, which will be displayed in VS Code directly.
Related to the PR, please use "VS Code", not "VSCode". We also ask that first mention in a topic (first paragraph) should be the full name "Visual Studio Code".

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (gregvanl) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:9]: Please use "Visual Studio Code" in the first mention.

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (gregvanl) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:3]: Please use "Visual Studio Code" in the description.

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (gregvanl) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:18]: Please use "VS Code", with a space, not "VSCode".

**Comment by sfoslund** [comment]
@gregvanl thanks for the feedback! Writing docs is one of our last steps before we ship a 1.0 version and take the prerelease tag off. If it makes sense to you we can keep this PR in a draft state until we finalize things, then publish this article when the 1.0 release is out.

**Comment by sfoslund** [comment]
@gregvanl @tdykstra we're going to release the 1.0 version any day now and the repo has been updated so it is no longer marked as a preview. Do you want me to retarget this PR to somewhere within the VSCode docs now or should we continue here?

**Comment by gregvanl** [comment]
@sfoslund I think you'll get better readership in the dotnet docs. We've found that .NET content gets lost on the VS Code website and most of those developers are doing either web or Python or C++ development. Especially for VS Code extension authors who are mostly using JavaScript and TypeScript.

**Comment by tdykstra** [comment]
We should add it then under Additional tools in the TOC, and a short section for it in the [Overview](https://docs.microsoft.com/en-us/dotnet/core/additional-tools/) for Additional tools

**Comment by tdykstra** [comment]
@sfoslund If you prefer, we can merge this PR and I'll make a separate follow-up PR for the TOC and overview changes.

**Comment by sfoslund** [comment]
No problem, I added this tool to the overview page. I wasn't able to find the table of contents though, can you point that file out to me?

**Comment by tdykstra** [comment]
https://github.com/dotnet/docs/blob/1f0bafb4ea3f450c82b9c54b2bf856ad63ebce22/docs/fundamentals/toc.yml#L348

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (tdykstra) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:2]:
```suggestion
title: .NET install tool for VS Code extension authors
```

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (tdykstra) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:3]:
```suggestion
description: An overview of the .NET install Tool for extension authors, a Visual Studio Code extension for installing the .NET runtime.
```

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (tdykstra) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:7]:
```suggestion
# .NET Install tool for extension authors
```

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (tdykstra) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:9]:
```suggestion
The [.NET Install tool for extension authors](https://github.com/dotnet/vscode-dotnet-runtime) is a Visual Studio Code extension that allows acquisition of the .NET runtime specifically for VS Code extension authors. This tool is intended to be leveraged in extensions that are written in .NET and require .NET to boot pieces of the extension (for example, a language server). The extension is not intended to be used directly by users to install .NET for development.
```

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (tdykstra) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:11]:
```suggestion
## Getting started: extension authors
```

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (tdykstra) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:13]:
```suggestion
To ensure that the .NET Install tool for extension authors is the right fit for your scenario, start by reviewing the [goals of this extension](https://github.com/dotnet/vscode-dotnet-runtime#goals-acquiring-net-core-for-extensions) on our GitHub page.
```

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (tdykstra) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:18]:
```suggestion
Once you have verified that the .NET Install tool for extension authors fits your needs, you can take a dependency on it in your [extension manifest](https://code.visualstudio.com/api/references/extension-manifest) and begin using the commands we expose with the [VS Code API](https://code.visualstudio.com/api/extension-guides/command#programmatically-executing-a-command). You can find the list of commands this extension exposes on our [GitHub](https://github.com/dotnet/vscode-dotnet-runtime/blob/master/Documentation/commands.md).
```

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (tdykstra) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:28]:
```suggestion
## Getting started: end users
```

File: docs/core/additional-tools/vscode-dotnet-runtime.md
Comment (tdykstra) [inline docs/core/additional-tools/vscode-dotnet-runtime.md:30]:
```suggestion
In general, the end user should not need to interact with the .NET Install tool for extension authors at all. If you are having problems with the extension, check out our [troubleshooting page](https://github.com/dotnet/vscode-dotnet-runtime/blob/master/Documentation/troubleshooting.md) or file an issue on our [GitHub](https://github.com/dotnet/vscode-dotnet-runtime/issues).
```

File: docs/fundamentals/toc.yml
Comment (tdykstra) [inline docs/fundamentals/toc.yml:370]:
```suggestion
    - name: .NET Install tool for extension authors
```

**Comment by sfoslund** [comment]
@tdykstra great, should be good to go now.

File: docs/core/additional-tools/index.md
Comment (tdykstra) [inline docs/core/additional-tools/index.md:26]:
```suggestion
## .NET Install tool for extension authors
```

File: docs/core/additional-tools/index.md
Comment (tdykstra) [inline docs/core/additional-tools/index.md:28]:
```suggestion
The [.NET Install tool for extension authors](https://github.com/dotnet/vscode-dotnet-runtime) is a Visual Studio Code extension that allows acquisition of the .NET Core runtime specifically for VS Code extension authors. This tool is intended to be leveraged in extensions that are written in .NET and require .NET to boot pieces of the extension (for example, a language server). The extension is not intended to be used directly by users to install .NET for development.
```

**Review by tdykstra — APPROVED** [review]
I left suggestions for changing titles/headings to sentence case. Once those are addressed are we ready to merge?

**Comment by sfoslund** [comment]
> Once those are addressed are we ready to merge?

Yes, I'm happy to merge!

**Comment by sfoslund** [comment]
Thanks @tdykstra!

**Pull request merged**
